### PR TITLE
Switch type literals to use builtin functions.

### PR DESCRIPTION
**Pull request by jonmeow**
Modifies the core package and literal handling to use factory functions for standard type literals.

Updates function/builtin/import.carbon to stop depending on the prelude, since it would list all the impls in the core file. Updates alias/builtins.carbon to be failing (the type values cannot be aliased).

### CODE REVIEW

**Comment by jonmeow** [comment]
Depends on #3977

File: core/prelude.carbon [hunk 1]
```diff
@@ -6,11 +6,6 @@
 
 package Core library "prelude";
 
+export import library "prelude/types/bool";
 export import library "prelude/operators";
 export import library "prelude/types";
```
Comment (zygoloid) [inline core/prelude.carbon:10]: Should "prelude/types" export "prelude/types/bool"?

**Review by zygoloid — APPROVED** [review]
Looks good. I wonder how much all these extra function calls will slow down @chandlerc's performance benchmark.

**Comment by jonmeow** [comment]
> Looks good. I wonder how much all these extra function calls will slow down @chandlerc's performance benchmark.

Noted this on #toolchain, but I think it's WAI

File: core/prelude.carbon (hunk 1 above)
Comment (jonmeow) [reply core/prelude.carbon:10]: Done

**Pull request merged**